gravitee-am-test/specs/gateway/reset-password.jest.spec.ts

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import fetch from "cross-fetch";
+import {afterAll, beforeAll, expect, jest} from "@jest/globals";
+import {requestAdminAccessToken} from "@management-commands/token-management-commands";
+import {createDomain, deleteDomain, patchDomain, startDomain} from "@management-commands/domain-management-commands";
+import {createApplication, updateApplication} from "@management-commands/application-management-commands";
+import {
+    extractXsrfTokenAndActionResponse,
+    getWellKnownOpenIdConfiguration,
+    performFormPost,
+    performGet
+} from "@gateway-commands/oauth-oidc-commands";
+import cheerio from "cheerio";
+import {createUser} from "@management-commands/user-management-commands";
+import {clearEmails, getLastEmail} from "@utils-commands/email-commands";
+
+global.fetch = fetch;
+
+let domain;
+let accessToken;
+let application;
+let openIdConfiguration;
+let user;
+let confirmationLink;
+
+jest.setTimeout(200000)
+
+beforeAll(async () => {
+    const adminTokenResponse = await requestAdminAccessToken();
+    accessToken = adminTokenResponse.body.access_token;
+    domain = await createDomain(accessToken, "test-reset-password", "test Gateway reset password with password history");
+});
+
+describe("Gateway reset password", () => {
+    beforeAll(async () => {
+        domain = await patchDomain(domain.id, accessToken, {
```

Comment:
PasswordPolicy can be overided at application level.
In this simulation we only check that the domain settings is working but IMHO we should also have a dedicated test to validate the usage of the application settings.

In both case we should check that the first password can be reused after 3 resetPassword.

Reply:
Done. Application stuff can be done during the quality sprint.